File: 2021/src/test/java/nel/marco/Day7Test.java
package nel.marco;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Day7Test {

    Day7 day7 = new Day7();

    public List<Integer> readInput(boolean readExample) throws IOException {

        String path = "src\\main\\resources\\day7";
        if (readExample) {
            path = "src\\main\\resources\\day7Example";
        }

        List<String> stringList = Files.readAllLines(Path.of(path));

        return new ArrayList<>(Arrays.stream(stringList.get(0).split(",")).toList().stream().map(Integer::valueOf).toList());

    }

    @Test
    void part1() throws IOException {

        assertEquals(37, day7.part1(readInput(true)));
        assertEquals(355592, day7.part1(readInput(false)));

    }

    @Test
    void part2() throws IOException {

        assertEquals(168, day7.part2(readInput(true)));
        assertEquals(101618069, day7.part2(readInput(false)));
    }


}

File: 2015/src/test/kotlin/Day7Test.kt
import org.junit.jupiter.api.Test

import org.junit.jupiter.api.Assertions.*

internal class Day7Test {

    @Test
    fun part1() {
        val toList = """
            123 -> x
            456 -> y
            x AND y -> d
            x OR y -> e
            x LSHIFT 2 -> f
            y RSHIFT 2 -> g
            NOT x -> h
            NOT y -> i
        """.trimIndent()
            .split("\n")
            .toList()


        val day7 = Day7(toList)
        kotlin.test.assertEquals(123, day7.part1())
    }

    @Test
    fun part2() {
    }
}